src/main/cpp/src/SparkResourceAdaptorJni.cpp

```diff
@@ -226,6 +226,24 @@ struct task_metrics {
   }
 };
 
+enum class oom_type {
+  CPU_OR_GPU = 0,
+  CPU,
+  GPU,
+};
+
+struct oom_state_type {
+  int hit_count   = 0;
+  int skip_count  = 0;
+  oom_type filter = oom_type::CPU_OR_GPU;
+
+  bool matches(bool is_for_cpu)
+  {
+    return (is_for_cpu && (filter == oom_type::CPU || filter == oom_type::CPU_OR_GPU)) ||
+           (!is_for_cpu && (filter == oom_type::GPU || filter == oom_type::CPU_OR_GPU));
```

Comment:
nit, slightly cleaner? 
```suggestion
    return filter == oom_type::CPU_OR_GPU || 
           is_for_cpu && filter == oom_type::CPU ||
           !is_for_cpu && filter == oom_type::GPU;
```